src/riak_kv_ts_util.erl

```diff
@@ -155,16 +155,22 @@ do_epoch(String, CompleteFun, Exponent) ->
 
 maybe_convert_to_epoch({'EXIT', _}, Value, _Op) ->
     {binary, Value};
-maybe_convert_to_epoch(timestamp, Value, '>') ->
-    %% For strictly greater than, we increment any incomplete
-    %% date/time strings by one "unit", where the unit is the least
-    %% significant value provided by the user. "2016" becomes
-    %% "2017". "2016-07-05T11" becomes "2016-07-05T12".
+maybe_convert_to_epoch(timestamp, Value, Op) when Op == '>'; Op == '<=' ->
+    %% For strictly greater-than or less-than-equal-to, we increment
+    %% any incomplete date/time strings by one "unit", where the unit
+    %% is the least significant value provided by the user. "2016"
+    %% becomes "2017". "2016-07-05T11" becomes "2016-07-05T12".
+    %%
+    %% The result is then decremented by 1ms to maintain the proper
+    %% semantics. The alternative would be to convert the operator in
+    %% the syntax tree (from > to >=, or from <= to <).
```

Comment:
The alternative you are offering to consider would, hands down, be more efficient (I'm not willing to suggest this needs to be implemented now, though).